examples/albef/finetune_retrieval.py

```diff
@@ -0,0 +1,304 @@
+        _, topk_idx = sims.topk(k, dim=0)
+        score = model(
+            image=image_embeds[topk_idx],
+            text=text_embeds[start + i].repeat(k, 1, 1),
+            text_atts=text_atts[start + i].repeat(k, 1, 1),
+            input_type="multimodal",
+            is_train=False,
+        )
+        text_to_image_scores[start + i, topk_idx] = score
+
+        if i % log_every_n_steps == 0:
+            total_time = time.time() - start_time
+            time_str = "time {},".format(datetime.timedelta(seconds=int(total_time)))
+            batch_str = "batch {}/{},".format(i, len(sims_matrix[start:end]))
+            print("text to image retrieval", time_str, batch_str)
+
+    if dist_utils.is_dist_avail_and_initialized():
+        dist.barrier()
+        torch.distributed.all_reduce(
+            image_to_text_scores, op=torch.distributed.ReduceOp.SUM
+        )
+        torch.distributed.all_reduce(
+            text_to_image_scores, op=torch.distributed.ReduceOp.SUM
+        )
+
+    return image_to_text_scores.cpu(), text_to_image_scores.cpu()
+
+
+@torch.no_grad()
+def itm_eval(
+    image_to_text_scores,
+    text_to_image_scores,
+    image_to_text_mapping,
+    text_to_image_mapping,
+):
+    # Images to Text
+    ranks = torch.zeros(image_to_text_scores.size(0))
+    for index, score in enumerate(image_to_text_scores):
+        inds = torch.flip(torch.argsort(score), dims=[0])
+        rank = 1e10
+        # each image has multiple text mappings
+        # check retrieved inds with each ground truth mappping i
+        for i in image_to_text_mapping[index]:
+            tmp = torch.where(inds == i)[0][0]
+            if tmp < rank:
+                rank = tmp
+        ranks[index] = rank
+
+    # Compute metrics
+    tr1 = 100.0 * len(torch.where(ranks < 1)[0]) / len(ranks)
+    tr5 = 100.0 * len(torch.where(ranks < 5)[0]) / len(ranks)
+    tr10 = 100.0 * len(torch.where(ranks < 10)[0]) / len(ranks)
+
+    # Text to Images
+    ranks = torch.zeros(text_to_image_scores.size(0))
+    for index, score in enumerate(text_to_image_scores):
+        inds = torch.flip(torch.argsort(score), dims=[0])
+        ranks[index] = torch.where(inds == text_to_image_mapping[index])[0][0]
+
+    # Compute metrics
+    ir1 = 100.0 * len(torch.where(ranks < 1)[0]) / len(ranks)
+    ir5 = 100.0 * len(torch.where(ranks < 5)[0]) / len(ranks)
+    ir10 = 100.0 * len(torch.where(ranks < 10)[0]) / len(ranks)
+
+    tr_mean = (tr1 + tr5 + tr10) / 3
+    ir_mean = (ir1 + ir5 + ir10) / 3
+    r_mean = (tr_mean + ir_mean) / 2
+
+    eval_result = {
+        "txt_r1": tr1,
+        "txt_r5": tr5,
+        "txt_r10": tr10,
+        "txt_r_mean": tr_mean,
+        "img_r1": ir1,
+        "img_r5": ir5,
+        "img_r10": ir10,
+        "img_r_mean": ir_mean,
+        "r_mean": r_mean,
+    }
+    return eval_result
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", default="./examples/albef/configs/retrieval.yaml")
```

Comment:
Will this default work if run from a different directory? If not, maybe you can use os.getcwd() or something similar

Reply:
do you want to correct this yourself? Just to flag this. Your call.